src/Sentry.Unity/SentryUnityOptions.cs

```diff
@@ -175,7 +176,10 @@ internal SentryUnityOptions(SentryMonoBehaviour behaviour, IApplication applicat
                 ? "editor"
                 : "production";
 
-            CacheDirectoryPath = application.PersistentDataPath.Trim();
+            if (application.Platform is not RuntimePlatform.Switch)
```

Comment:
The first stab at this was through the `UnknownPlatform.cs`, for any platform that's not explicitly known to work. If the issue is with creating a directory, I'd suggest moving [that code](https://github.com/getsentry/sentry-unity/blob/f2c2d3aeda3afe94ca5bac5941e845c02b6f6b94/src/Sentry.Unity/Default/UnknownPlatform.cs#L24_L29) here instead

Reply:
Fair point. I think we can use `ToSentryOptions` from now on for more than just mapping from the `ScriptableOptions`to the `SentryOptions`.